### ISSUE TITLE
React hook form getValues doesn't get updated for optional fields

### ISSUE THREAD

**Issue by tanmay-patil** [react-hook-form/react-hook-form#8686]
### Version Number

7.33.1

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-v7-usewatch-forked-xsu35f

### Steps to reproduce

1. Enter details for all 3 fields
2. Notice when user enters text for middle name, the field data isn't reflected in the getValues() which is console logged.
3. This is because middle name field happens to be optional field, and firstname is a required field.
4. This issue occurs primarily when the form contains mix of required and optional fields, where the optional fields data is never updated into the getValues()

### Expected behaviour

1. The optional fields changes should be reflected properly into the getValues() of the form.

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
You want to use the `watch` / `useWatch` API to subscribe to the value change. `getValues` doesn't "react" to changes by design. It's purpose is to retrieve the current form values, say, in a function without a re-render.

The reason it works for you when you first enter input values is that the `blur` event fires and the form validation state is updated, causing a re-render, hence the "fresh" values in your render.

TLDR; if you want reactive values, use the `watch` / `useWatch`

**Comment by Moshyfawn** [comment]
To optimize render count, you can move the form values "display" to it's own component and `useWatch` in it to localize the renders.
```js
const Display = ({ control }) => {
  const values = useWatch({ control })
  
  return JSON.stringify(values)
}

const Form = () => {
  const { control } = useForm()
  
  return (
    <form>
      <input {...register("field")} />
      <Display control={control} />
    </form>
  )
}
```

**Comment by tanmay-patil** [comment]
@Moshyfawn Thanks a lot! This helps
